lib/metrics/dial_failure.go

```diff
@@ -0,0 +1,13 @@
+package metrics
+
+import (
+	"github.com/status-im/telemetry/pkg/types"
+)
+
+type DialFailure struct {
+	GenericMetric[types.DialFailure]
+}
+
+var (
+	DialFailureProcessor = &DialFailure{}
+)
```

Comment:
This is missing the implementation isn't it?

Reply:
Nope! After my generic metrics PR this is all that is necessary to add all the db logic 😄 https://github.com/status-im/telemetry/pull/56/files#diff-c4d76f1754d2a1b8e04e65bde83777fa2509c22740f0dff2d8e8a7efcb19ae3b

Reply:
All the column names and values necessary to create the insert query are derived based on the json attributes of the `DialFailure` type. As long as the name of the field matches the name of the column, it works. https://github.com/status-im/telemetry/blob/53fa8accec5d37e2899314a160702a5909eb2948/lib/metrics/generic.go#L103

Reply:
Oh, right, I completely forgot. That's awesome!